Allow the user more control over error handling

ismyrnow opened the pull request:
Instead of returning `res.send(401)`, call `authError()`, and allow the user to handle the error.

The use case for this is to allow the user to throw a custom error in the `verify` or `lookup` functions, and handle that error, possibly sending the message back with the response. See the new test for an example of this.

This would be a major version change, since it's possible code may be relying on the previous `res.send(401)` calls, and isn't using the default `authError` function.

Commit:
Add gitignore

Commit:
Add failing test for error propogation

Commit:
Call authError if anything goes wrong

- Instead of returning `res.send(401)`, call `authError()`, and allow the user to handle the error.

calvinmetcalf commented:
You should just set up a global gitignore, I only include them in libs for
things specific to the lib.

As it is this is a breaking change we probably want different functions for
the verify middle ware vs the token endpoints because errors mean very
different things in the 2 places.

On Wed, Apr 15, 2015, 5:31 PM Ishmael Smyrnow notifications@github.com
wrote:

> Instead of returning res.send(401), call authError(), and allow the user
> to handle the error.
> 
> The use case for this is to allow the user to throw a custom error in the
> verify or lookup functions, and handle that error, possibly sending the
> message back with the response. See the new test for an example of this.
> 
> This would be a major version change, since it's possible code may be
> relying on the previous res.send(401) calls, and isn't using the default
> 
> ## authError function.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/calvinmetcalf/express-simple-auth-token/pull/1
> Commit Summary
> - Add gitignore
> - Add failing test for error propogation
> - Call authError if anything goes wrong
> 
> File Changes
> - _A_ .gitignore
>   https://github.com/calvinmetcalf/express-simple-auth-token/pull/1/files#diff-0
>   (2)
> - _M_ index.js
>   https://github.com/calvinmetcalf/express-simple-auth-token/pull/1/files#diff-1
>   (24)
> - _M_ test.js
>   https://github.com/calvinmetcalf/express-simple-auth-token/pull/1/files#diff-2
>   (22)
> 
> Patch Links:
> 
>    -
>    https://github.com/calvinmetcalf/express-simple-auth-token/pull/1.patch
>    -
>    https://github.com/calvinmetcalf/express-simple-auth-token/pull/1.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/calvinmetcalf/express-simple-auth-token/pull/1.

Commit:
Error functions for create and refresh token routes

Commit:
Update readme

- Includes a smaller usage example at the top, and updated formatting for the settings.

ismyrnow commented:
Updated the PR. There are now two new error functions: `createTokenError` and `refreshTokenError`. This commit is backwards compatible, and provides more flexibility in handling errors.

The readme has also been updated with the new settings.

Commit:
Remove gitignore
